### PR TITLE
Remove iAd (Apple Search Ads) attribution

### PR DESCRIPTION
**Pull request by jkmassel**
We're not purchasing ads for the app, and this code is deprecated. Let's remove it for now.

Includes some Xcode project cleanup noise

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
LGTM. 

It seems that this API is not only deprecated but is also no longer working:

> 
> After February 7, 2023, all requests made to the Apple Search Ads iAd Attribution API will return with a value of "iad-attribution" = false, or errors.  See ADClient/requestAttributionDetails(_:).
>
> Source: https://developer.apple.com/documentation/iad/setting-up-apple-search-ads-attribution

There was a relatively recent search ad-campaign, but I'm not sure if it used the attribution info. It's reasonable to assume that if we run one again, we'll need to add a new attribution tracking code.

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -19553,7 +19541,7 @@
 				bg,
 				sk,
 			);
-			mainGroup = 29B97314FDCFA39411CA2CEA;
```
Comment (kean) [inline WordPress/WordPress.xcodeproj/project.pbxproj:19556]: > Includes some Xcode project cleanup noise

I think it's coming from your Xcode version. Have you updated to Xcode 15.4? A PR landed recently https://github.com/wordpress-mobile/WordPress-iOS/pull/23259, but I'm still on 15.1.

Comment (kean) [reply WordPress/WordPress.xcodeproj/project.pbxproj:19556]: Oh, I see that we now fully switched to 15.4 – https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.xcode-version. I tried it and I'm getting the same diff after making any pbxproj changes. Resolving.

**Review by kean — APPROVED** [review]

**Pull request merged**